### DISCUSSION TITLE
Need to be able to reorder and rename checkout form field

### DISCUSSION THREAD

**Discussion by FinnAr** [woocommerce/woocommerce#52719] [Checkout Flow]
To avoid adding dependencies to code snippets, it would be great if it was possible to modify the order and the labels of the checkout form fields via configuration options.

**Comment by nielslange** [comment]
Hello @FinnAr 👋

Thank you for your question! To better assist you, could you please specify whether you're using the shortcode-based checkout or the Checkout block? Additionally, it would be helpful to know which fields you'd like to reorder and what you envision for the final layout of these fields. Are there specific countries or situations where you expect these changes to apply?

**Comment by FinnAr** [comment]
Hello Niels

 

I’m a newcomer to WooCommerce, and I’m using the latest version, so I’m pretty sure I’m using the Checkout block.   I would like to rename the Company Name field and move it to the top of the block.   This would be a permanent change with no exceptions.

**Comment by nielslange** [comment]
> I’m new to WooCommerce and using the latest version, so I believe I’m working with the Checkout block. I want to rename the "Company Name" field and move it to the top of the block. This change should be permanent and apply without exceptions.

@FinnAr Thanks for clarifying! Since the country selection is crucial for determining payment and shipping options, we’ve positioned the country field at the top of the billing and shipping address forms. To avoid negatively impacting your revenue, we recommend keeping the country field at the top. 

However, if it's important for you to move the "Company Name" field to the top, you can use one of the following code snippets:

```php
// Rename the company field and place it at the top for the Netherlands
add_filter(
    'woocommerce_get_country_locale',
    function ( $locale ) {
        $locale['NL']['country']['priority'] = 2;
        $locale['NL']['company']['label']    = __( 'Business', 'YOUR-TEXTDOMAIN' );
        $locale['NL']['company']['priority'] = 1;

        return $locale;
    }
);
```

```php
// Rename the company field and place it at the top for all countries
add_filter(
    'woocommerce_get_country_locale',
    function ( $locales ) {
        foreach ( $locales as $country_code => &$locale ) {
            $locale['country']['priority'] = 2;
            $locale['company']['label']    = __( 'Business', 'YOUR-TEXTDOMAIN' );
            $locale['company']['priority'] = 1;
        }
        return $locales;
    }
);
```

You can add these snippets to your site using the [Code Snippets](https://wordpress.org/plugins/code-snippets/) plugin. These snippets rename the "Company Name" field and move it to the top of the form.

> [!NOTE]
> The `'label' => __( 'Business', 'YOUR-TEXTDOMAIN' )` line ensures the label can be translated. For more information on internationalization, check out [How to Internationalize Your Plugin](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/) and [Internationalization for Themes](https://developer.wordpress.org/themes/functionality/internationalization/). If your site is only in one language, you can replace this line with `'label' => 'Business'`, or your desired string, for simplicity.

**Comment by FinnAr** [comment]
Hi Niels

 

Thanks for responding so quickly.

 

Sorry about the confusion; no, I agree the country selection should be at top.  However, I need a field before the name and address section.  This could be a rename and reordering of the company name field, or it could be a new custom field.

 

Finn

**Comment by nielslange** [comment]
> Sorry about the confusion; no, I agree the country selection should be at top.  However, I need a field before the name and address section.  This could be a rename and reordering of the company name field, or it could be a new custom field.

To ensure we avoid unnecessary extra work, could you please provide more details about what you're looking for and describe the expected result?

**Comment by FinnAr** [comment]
Hi again

 

Certainly. 😊

 

It would be great if it were possible to rename and reorder fields in the blocks on the checkout page.  An added benefit would be to be able to add custom fields.

 

As for my case, I need a new required field above the Billing Name (don’t remember if Name is shown as a First Name and a Family Name field).   This could be a new custom field, or it could be the Company Name field that is renamed and moved up before the Billing Name line.

 

The expected result would be a new required field before the Billing Name line.

**Comment by nielslange** [comment]
> It would be great if it were possible to rename and reorder fields in the blocks on the checkout page.  An added benefit would be to be able to add custom fields.

To rename a field, simply overwrite the `label` value of the corresponding field. The snippet, I shared in https://github.com/woocommerce/woocommerce/discussions/52719#discussioncomment-11302465, shows how to do that. Same counts for reordering fields. In that case, you need to adjust the `priority` value.

To add custom fields, you can use the [additional fields API](https://developer.woocommerce.com/docs/cart-and-checkout-additional-checkout-fields/). In https://github.com/woocommerce/woocommerce/discussions/51177#discussioncomment-11276189, I shared an example of how to add a VAT field to the checkout.

**Comment by FinnAr** [comment]
Many thanks, Niels!

Just one more question (probably silly, but I am a newbie):  When adding the snippet to code snippets as explained, I get an error message on the line

        foreach ( $locales as $country_code => &$locale ) {

where code snippets points to the > and says that “special characters must be escaped”.   What goes wrong here?

**Comment by nielslange** [comment]
> Just one more question (probably silly, but I am a newbie):  When adding the snippet to code snippets as explained, I get an error message on the line
> 
>         foreach ( $locales as $country_code => &$locale ) {
> 
> where code snippets points to the > and says that “special characters must be escaped”.   What goes wrong here?

It's hard to say what's causing this issue, as the code snippet is correct. It does work flawlessly on my end, and using `&$locale` instead of `$locale`, is required for the script to work correctly. 

<img width="2560" alt="Screenshot 2024-11-26 at 11 50 24" src="https://github.com/user-attachments/assets/cf9b89d4-36fd-4800-922a-9d18d58a2508">

I suggest reaching out to the developer of the Code snippets plugin via [their support channel](https://wordpress.org/support/plugin/code-snippets/). It might also be related to your host. Thus, you could also contact them. I hope you;ll be able to sort this out.

**Comment by FinnAr** [comment]
Many thanks for your help and patience, Niels!

 

After installing an update to Code Snippets, the error message disappeared, and I was able to Save and Activate the snippet, which is identical to the one you have in your screenshot except for the characters in the label text.

I configured the snippet to “Run Snippet everywhere”.

 

However, I don’t seem to see any effect on the checkout page, which still looks like this: ☹    Anything else I may have missed?

**Comment by FinnAr** [comment]
Wait a minute – it works now … 😊

 

From: Finn Arildsen ***@***.***> 
Sent: 28. november 2024 12:42
To: 'woocommerce/woocommerce' ***@***.***>
Subject: RE: [woocommerce/woocommerce] Need to be able to reorder and rename checkout form field (Discussion #52719)

 

Many thanks for your help and patience, Niels!

 

After installing an update to Code Snippets, the error message disappeared, and I was able to Save and Activate the snippet, which is identical to the one you have in your screenshot except for the characters in the label text.

I configured the snippet to “Run Snippet everywhere”.

 

However, I don’t seem to see any effect on the checkout page, which still looks like this: ☹    Anything else I may have missed?

**Comment by nielslange** [comment]
> Wait a minute – it works now … 😊

I'm glad to hear that it finally works. 🙌
